### PR TITLE
Use ReTest.jl for tests

### PR DESCRIPTION
**Pull request by daniel-thom**
This PR adopts ReTest.jl in order to make it easier to run subsets of tests.

### CODE REVIEW

**Review by jd-lara — APPROVED** [review]

File: test/load_tests.jl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using Revise
```
Comment (GabrielKS) [inline test/load_tests.jl:1]: Seems like even if the best practice is to install Revise globally, if our tests are going to depend it we ought to have it in `test/Project.toml`

**Review by GabrielKS — COMMENTED** [review]
See my initial design-level comments here: https://github.com/NREL-Sienna/PowerSystems.jl/pull/1052#discussion_r1490153760

File: test/load_tests.jl (hunk 1 above)
Comment (jd-lara) [reply test/load_tests.jl:1]: This was an oversight. We shouldn't add Revise to the test env

Comment (GabrielKS) [reply test/load_tests.jl:1]: Revise definitely gets used in this PR though

Comment (jd-lara) [reply test/load_tests.jl:1]: The default run script for testing in CI doesn't use Revise

**Pull request merged**